### PR TITLE
Docs: Add json docs.

### LINKED ISSUE #3753: Documentation on JSON function modules is required.
### Details

There is no comprehensive documentation for the json functionality and code currently implemented, and this part of the documentation is required.

### PR DESCRIPTION
**Pull request by grysgreat**
Added some documents related to json in abacus.

### Linked Issue
Fix #3753

### What's changed?
Two json documents have been added in the docs/advanced directory.

- json_para: Describes the structure of the abacus.json file output by abacus and the meanings of each parameter.

- json_add: Describes the current implementation path of json code in abacus and how to use the current json addition function in abacus to build a json tree.

### A example abacus.json file:
- Run calculate case of `examples/relax/pw_al`
-  [abacus.json](https://github.com/deepmodeling/abacus-develop/files/14681929/abacus.2.json)

### CODE REVIEW

**Comment by grysgreat** [comment]
A example abacus.json file:
- Run calculate case of `examples/relax/pw_al`
-  [abacus.json](https://github.com/deepmodeling/abacus-develop/files/14681929/abacus.2.json)

**Review by WHUweiqingzhou — COMMENTED** [review]
I think these docs should be located at docs/advanced/output_files/.

**Comment by grysgreat** [comment]
> output_files

I think the json_add document focuses on how JSON-related code is used. Not exactly introduce output_files. How about move tnem into docs/advanced/json/.?

**Comment by WHUweiqingzhou** [comment]
> > output_files
> 
> I think the json_add document focuses on how JSON-related code is used. Not exactly introduce output_files. How about move tnem into docs/advanced/json/.?

OK, second it.

File: docs/advanced/json/json_para.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+## Output
+
+An array of dicts, including information about each self-consistent field (SCF) step, such as energy, convergence, and configuration:
+
+- `energy`, `e_fermi` - [double] The total energy and Fermi energy. Unit in eV.
+- `force` -  [array(array(double))] The forces calculated on each atom. Unit in eV/Angstrom.
+- `stress` - [array(array(double))] The stress tensor. Unit in Kbar.
+- `cell` - [array(array(double))] The cell parameters. Unit in Angstrom.
+- `coordinate` - [array(array(double))] The coordinates of the atoms in the box after the simulation.
+- `total_mag` , `absolute_mag` - [double] The total magnetic moment, total absolute magnetic moment, and a list of magnetic moments for each atom, respectively.
```
Comment (pxlxingliang) [inline docs/advanced/json/json_para.md]: lack atom_mag?

Comment (grysgreat) [reply docs/advanced/json/json_para.md]: fix that~

**Review by pxlxingliang — APPROVED** [review]

**Pull request merged**